### ISSUE TITLE
Time conversion to/from Unix time is incorrect

### ISSUE THREAD

**Issue by avitkauskas** [vlang/v#2331]
**V version:** V 0.1.21 b1806b5
**OS:** Mac OS

Either `unix()` or `calc_unix()` functions are wrong.
Also, `unix()` function does not fill in `.uni` field.
As it is used in `add_seconds()`, this also makes `uni` field `0`.

**What did you do?**
```v
import time

d1 := time.parse('2020-01-01 00:00:00')
d2 := d1.add_seconds(time.seconds_per_day)
d3 := time.unix(d1.uni)

println(d1)
println(d2)
println(d3)
```

**What did you expect to see?**
```
{
	year: 2020
	month: 1
	day: 1
	hour: 0
	minute: 0
	second: 0
	uni: 1577829600
}
{
	year: 2020
	month: 1
	day: 2
	hour: 0
	minute: 0
	second: 0
	uni: 1577916000
}
{
	year: 2020
	month: 1
	day: 1
	hour: 0
	minute: 0
	second: 0
	uni: 1577829600
}
```
 
**What did you see instead?**
```
{
	year: 2020
	month: 1
	day: 1
	hour: 0
	minute: 0
	second: 0
	uni: 1577829600
}
{
	year: 2020
	month: 1
	day: 1
	hour: 22
	minute: 0
	second: 0
	uni: 0
}
{
	year: 2020
	month: 12
	day: 30
	hour: 22
	minute: 0
	second: 0
	uni: 0
}
```
`d2 := d1.add_seconds(time.seconds_per_day)` added only 22 hours, not a full day.
`d3 := time.unix(d1.uni)` did not convert date-time back correctly.

**Comment by medvednikov** [comment]
Thanks, `time` needs lots of love. `uni` needs to finally be replaced with `unix` as well.

**Comment by avitkauskas** [comment]
Made some investigation. The problem is that C function `mktime()` that is used in `parse()` via `new_time()` via `calc_unix()` to set the `.uni` field counts in the timezone:
https://pubs.opengroup.org/onlinepubs/009695399/functions/mktime.html
"Local timezone information shall be set as though mktime() called tzset()."
https://pubs.opengroup.org/onlinepubs/009695399/functions/tzset.html
But `unix()` interprets `.uni` as UTC (though it lokks like it still messes up the year and month calculation).

**Comment by avitkauskas** [comment]
Working on that. Will push a PR with proposed improvements to `time` module in a day or two (depending on my time).

**Comment by M4SSD35TRUCT10N** [comment]
@avitkauskas Do you work on that issue any longer? I've assigned you to this one as you mentioned you're working on it.

Your friendly issue janitor. ;-)

**Comment by avitkauskas** [comment]
@M4SSD35TRUCT10N No, I am not working on it. I was proposing a PR to fix the current issues with time, but there was a wish from comments to have it re-done in Go-like fashion. That would be much more work than I can afford.

**Comment by M4SSD35TRUCT10N** [comment]
@avitkauskas Okay. So, this is still an issue and should be fixed. I'll remove the assignment. Thank you for your response.

**Comment by alexesprit** [comment]
This issue was fixed in #3651.

Also, output provided in "What did you expect to see?" block is wrong. The correct output for 2020-01-01 00:00:00 date is the following:
```
{
        year: 2020
        month: 1
        day: 1
        hour: 0
        minute: 0
        second: 0
        v_unix: 1577836800
}
{
        year: 2020
        month: 1
        day: 2
        hour: 0
        minute: 0
        second: 0
        v_unix: 1577923200
}
{
        year: 2020
        month: 1
        day: 1
        hour: 0
        minute: 0
        second: 0
        v_unix: 1577836800
}
```

You can use any Unix time to date converter to verify Unix time is correct in the output I posted above.
